app/controllers/locations_controller.rb

```diff
@@ -5,7 +5,18 @@ class LocationsController < ApplicationController
 
   # GET /
   def index
-    @locations = Location.for_display
+    @locations =
+      if params[:search_in_bounds]
+        bounds = params[:search_in_bounds]
+        Location.in_bounds(bounds[:sw_lat], bounds[:sw_lng], bounds[:ne_lat], bounds[:ne_lng])
```

Comment:
This is now removing the `name <> ''` requirement.
I think this would read better if you create a default scope of `for_display` and then conditionally add the `in_bounds` constraint... then it could be something like:
```
scope = Location.for_display
scope = scope.in_bounds(params[:in_bounds]) if params[:in_bounds].present?
scope = scope.newest.limit(Setting.site_limit_locations)
@locations = scope
```
Maybe something like that?  You would need to change the scope to take named parameters, but that might be nicer ya?